conda/gateways/connection/session.py

```diff
@@ -68,19 +64,17 @@ def close(self):
         raise NotImplementedError()
 
 
-@lru_cache(None)
 def get_channel_name_from_url(url: str) -> str | None:
     """
     Given a URL, determine the channel it belongs to and return its name.
     """
-    channel_objects = get_channel_objs(context)
-
-    for channel in channel_objects:
+    for channel in get_channel_objs(context):
         for base_url in channel.base_urls:
             if url.lower().startswith(base_url):
                 return channel.canonical_name
 
 
+@lru_cache(None)
```

Comment:
adding caching here, since I don't think we need to construct this in-thread again and again, during the runtime of the CLI the channel settings won't change